File: flutter/pomodoro_timer/lib/features/settings/view/pomodoro_benefits_screen.dart
import 'package:flutter/material.dart';
import 'package:vibration/vibration.dart';

/// Educational screen explaining the Pomodoro Technique.
///
/// Matches the iOS app's PomodoroBenefitsView with:
/// - Origin story
/// - How it works (5 steps)
/// - Benefits
/// - Considerations
/// - Call-to-action
class PomodoroBenefitsScreen extends StatelessWidget {
  const PomodoroBenefitsScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return Scaffold(
      extendBodyBehindAppBar: true,
      appBar: AppBar(
        title: const Text('The Pomodoro Way'),
        centerTitle: true,
        backgroundColor: Colors.transparent,
        elevation: 0,
        surfaceTintColor: Colors.transparent,
      ),
      body: Container(
        color: Color.lerp(
          theme.scaffoldBackgroundColor,
          theme.colorScheme.primary,
          0.12,
        ),
        child: SafeArea(
          child: ListView(
            padding: const EdgeInsets.all(16),
            children: [
              // Header
              _buildHeader(theme),
            const SizedBox(height: 32),

            // History Section
            _buildHistorySection(theme),
            const SizedBox(height: 32),

            // How It Works
            _buildHowItWorksSection(theme),
            const SizedBox(height: 32),

            // Benefits
            _buildBenefitsSection(theme),
            const SizedBox(height: 32),

            // Considerations
            _buildConsiderationsSection(theme),
            const SizedBox(height: 32),

              // CTA
              _buildCTA(context, theme),
              const SizedBox(height: 24),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildHeader(ThemeData theme) {
    return Column(
      children: [
        Container(
          width: 100,
          height: 100,
          decoration: BoxDecoration(
            shape: BoxShape.circle,
            gradient: LinearGradient(
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              colors: [
                theme.colorScheme.primary.withValues(alpha: 0.2),
                theme.colorScheme.secondary.withValues(alpha: 0.15),
              ],
            ),
            boxShadow: [
              BoxShadow(
                color: theme.colorScheme.primary.withValues(alpha: 0.2),
                blurRadius: 20,
                offset: const Offset(0, 8),
              ),
            ],
          ),
          child: Icon(Icons.timer, size: 44, color: theme.colorScheme.primary),
        ),
        const SizedBox(height: 16),
        Text(
          'The Power of Pomodoro',
          style: theme.textTheme.headlineMedium?.copyWith(
            fontWeight: FontWeight.bold,
          ),
          textAlign: TextAlign.center,
        ),
        const SizedBox(height: 8),
        Text(
          'Focus deeper. Work smarter. Rest better.',
          style: theme.textTheme.titleMedium?.copyWith(
            color: theme.colorScheme.onSurfaceVariant,
          ),
          textAlign: TextAlign.center,
        ),
      ],
    );
  }

  Widget _buildHistorySection(ThemeData theme) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        _buildSectionHeader(
          icon: Icons.book,
          title: 'The Origin Story',
          color: theme.colorScheme.primary,
        ),
        const SizedBox(height: 16),
        Card(
          color: theme.cardColor,
          elevation: 0,
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(12),
            side: BorderSide(
              color: theme.dividerColor.withValues(alpha: 0.3),
              width: 0.5,
            ),
          ),
          child: Padding(
            padding: const EdgeInsets.all(16),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const Text(
                  'The Pomodoro Technique was created in the late 1980s by Francesco Cirillo, an Italian university student seeking a better way to study and manage his time.',
                ),
                const SizedBox(height: 12),
                const Text(
                  'Named after the tomato-shaped kitchen timer (pomodoro means "tomato" in Italian) he used to track his work intervals, this simple yet powerful method has since helped millions worldwide achieve better focus and productivity.',
                ),
                const SizedBox(height: 12),
                Row(
                  children: [
                    const Icon(Icons.lightbulb, color: Colors.orange, size: 20),
                    const SizedBox(width: 8),
                    Expanded(
                      child: Text(
                        'Born from necessity, perfected through practice',
                        style: theme.textTheme.bodyMedium?.copyWith(
                          fontStyle: FontStyle.italic,
                        ),
                      ),
                    ),
                  ],
                ),
              ],
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildHowItWorksSection(ThemeData theme) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        _buildSectionHeader(
          icon: Icons.settings,
          title: 'How It Works',
          color: Colors.blue,
        ),
        const SizedBox(height: 8),
        Text(
          'The technique is simple, sustainable, and scientifically backed:',
          style: theme.textTheme.bodyMedium?.copyWith(
            color: theme.colorScheme.onSurfaceVariant,
          ),
        ),
        const SizedBox(height: 16),
        _buildStepCard(
          number: 1,
          title: 'Choose Your Task',
          description: 'Select a single task or project you want to focus on.',
          icon: Icons.check_circle,
          color: theme.colorScheme.primary,
        ),
        const SizedBox(height: 12),
        _buildStepCard(
          number: 2,
          title: 'Set the Timer',
          description: 'Start a 25-minute focus session (one "pomodoro").',
          icon: Icons.timer,
          color: theme.colorScheme.primary,
        ),
        const SizedBox(height: 12),
        _buildStepCard(
          number: 3,
          title: 'Work Without Distraction',
          description: 'Focus entirely on your task until the timer rings.',
          icon: Icons.psychology,
          color: theme.colorScheme.primary,
        ),
        const SizedBox(height: 12),
        _buildStepCard(
          number: 4,
          title: 'Take a Short Break',
          description: 'Rest for 5 minutes. Stretch, hydrate, or relax.',
          icon: Icons.coffee,
          color: Colors.green,
        ),
        const SizedBox(height: 12),
        _buildStepCard(
          number: 5,
          title: 'Repeat & Recharge',
          description: 'After 4 pomodoros, take a longer 15-30 minute break.',
          icon: Icons.refresh,
          color: Colors.blue,
        ),
      ],
    );
  }

  Widget _buildBenefitsSection(ThemeData theme) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        _buildSectionHeader(
          icon: Icons.star,
          title: 'Why It Works',
          color: Colors.orange,
        ),
        const SizedBox(height: 8),
        Text(
          'Research and millions of users have found these key benefits:',
          style: theme.textTheme.bodyMedium?.copyWith(
            color: theme.colorScheme.onSurfaceVariant,
          ),
        ),
        const SizedBox(height: 16),
        _buildBenefitCard(
          icon: Icons.visibility,
          title: 'Enhanced Focus',
          description:
              'Short, timed sessions help maintain deep concentration and prevent mental drift.',
          color: theme.colorScheme.primary,
        ),
        const SizedBox(height: 12),
        _buildBenefitCard(
          icon: Icons.battery_full,
          title: 'Reduced Mental Fatigue',
          description:
              'Regular breaks prevent burnout and keep your mind fresh throughout the day.',
          color: Colors.green,
        ),
        const SizedBox(height: 12),
        _buildBenefitCard(
          icon: Icons.schedule,
          title: 'Better Time Awareness',
          description:
              'Learn to accurately estimate how long tasks take and plan more effectively.',
          color: Colors.blue,
        ),
        const SizedBox(height: 12),
        _buildBenefitCard(
          icon: Icons.trending_up,
          title: 'Increased Motivation',
          description:
              'Small wins and completed sessions create positive momentum and build confidence.',
          color: Colors.purple,
        ),
        const SizedBox(height: 12),
        _buildBenefitCard(
          icon: Icons.directions_walk,
          title: 'Sustainable Work Rhythm',
          description:
              'Balance focused work with restorative breaks for long-term productivity.',
          color: Colors.cyan,
        ),
        const SizedBox(height: 12),
        _buildBenefitCard(
          icon: Icons.back_hand,
          title: 'Distraction Management',
          description:
              'The commitment to 25 minutes helps you defer interruptions and stay on track.',
          color: Colors.orange,
        ),
      ],
    );
  }

  Widget _buildConsiderationsSection(ThemeData theme) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        _buildSectionHeader(
          icon: Icons.warning_amber,
          title: 'Things to Keep in Mind',
          color: Colors.orange,
        ),
        const SizedBox(height: 16),
        Card(
          color: theme.cardColor,
          elevation: 0,
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(12),
            side: BorderSide(
              color: theme.dividerColor.withValues(alpha: 0.3),
              width: 0.5,
            ),
          ),
          child: Padding(
            padding: const EdgeInsets.all(16),
            child: Column(
              children: [
                _buildConsiderationRow(
                  icon: Icons.account_tree,
                  text:
                      'May interrupt deep creative flow states that benefit from longer unbroken periods.',
                ),
                const Divider(height: 24),
                _buildConsiderationRow(
                  icon: Icons.calendar_today,
                  text:
                      'The rigid structure might not suit all task types or work environments.',
                ),
                const Divider(height: 24),
                _buildConsiderationRow(
                  icon: Icons.people,
                  text:
                      'Collaborative work may require flexibility with timing to accommodate others.',
                ),
                const Divider(height: 24),
                _buildConsiderationRow(
                  icon: Icons.tune,
                  text:
                      'Feel free to adjust session lengths to find what works best for you!',
                ),
              ],
            ),
          ),
        ),
        const SizedBox(height: 12),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 4),
          child: Text(
            'The Pomodoro Technique is a tool, not a rule. Adapt it to your needs and workflow.',
            style: theme.textTheme.bodySmall?.copyWith(
              color: theme.colorScheme.onSurfaceVariant,
              fontStyle: FontStyle.italic,
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildCTA(BuildContext context, ThemeData theme) {
    return Column(
      children: [
        Text(
          'Ready to experience the benefits?',
          style: theme.textTheme.titleLarge?.copyWith(
            fontWeight: FontWeight.w600,
          ),
          textAlign: TextAlign.center,
        ),
        const SizedBox(height: 16),
        FilledButton.icon(
          onPressed: () {
            Vibration.vibrate(duration: 50);
            Navigator.of(context).popUntil((route) => route.isFirst);
          },
          icon: const Icon(Icons.play_arrow),
          label: const Text('Start Your First Pomodoro'),
          style: FilledButton.styleFrom(
            padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 16),
            minimumSize: const Size(double.infinity, 56),
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(16),
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildSectionHeader({
    required IconData icon,
    required String title,
    required Color color,
  }) {
    return Row(
      children: [
        Icon(icon, color: color, size: 24),
        const SizedBox(width: 12),
        Text(
          title,
          style: TextStyle(
            fontSize: 24,
            fontWeight: FontWeight.bold,
            color: color,
          ),
        ),
      ],
    );
  }

  Widget _buildStepCard({
    required int number,
    required String title,
    required String description,
    required IconData icon,
    required Color color,
  }) {
    return Card(
      child: Padding(
        padding: const EdgeInsets.all(12),
        child: Row(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Container(
              width: 40,
              height: 40,
              decoration: BoxDecoration(
                shape: BoxShape.circle,
                color: color.withValues(alpha: 0.15),
              ),
              child: Center(
                child: Text(
                  '$number',
                  style: TextStyle(
                    fontSize: 18,
                    fontWeight: FontWeight.bold,
                    color: color,
                  ),
                ),
              ),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Row(
                    children: [
                      Icon(icon, size: 16, color: color),
                      const SizedBox(width: 6),
                      Text(
                        title,
                        style: const TextStyle(
                          fontSize: 17,
                          fontWeight: FontWeight.w600,
                        ),
                      ),
                    ],
                  ),
                  const SizedBox(height: 4),
                  Text(
                    description,
                    style: TextStyle(fontSize: 15, color: Colors.grey[600]),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildBenefitCard({
    required IconData icon,
    required String title,
    required String description,
    required Color color,
  }) {
    return Card(
      child: Padding(
        padding: const EdgeInsets.all(12),
        child: Row(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Container(
              width: 44,
              height: 44,
              decoration: BoxDecoration(
                borderRadius: BorderRadius.circular(10),
                color: color.withValues(alpha: 0.15),
              ),
              child: Icon(icon, color: color, size: 24),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    title,
                    style: const TextStyle(
                      fontSize: 17,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                  const SizedBox(height: 4),
                  Text(
                    description,
                    style: TextStyle(fontSize: 15, color: Colors.grey[600]),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildConsiderationRow({
    required IconData icon,
    required String text,
  }) {
    return Row(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Icon(icon, color: Colors.orange, size: 20),
        const SizedBox(width: 12),
        Expanded(child: Text(text, style: const TextStyle(fontSize: 15))),
      ],
    );
  }
}

File: native_apps/android/app/src/main/java/com/avtanshgupta/mr/pomodoro/ui/screens/benefits/PomodoroBenefitsScreen.kt
package avtanshgupta.PomodoroTimer.ui.screens.benefits

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import avtanshgupta.PomodoroTimer.ui.theme.PomodoroTheme

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PomodoroBenefitsScreen(
    onBackClick: () -> Unit,
    onGetStartedClick: () -> Unit
) {
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("The Pomodoro Way") },
                navigationIcon = {
                    IconButton(onClick = onBackClick) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Back"
                        )
                    }
                }
            )
        }
    ) { paddingValues ->
        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
        ) {
            // Background gradient
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .background(
                        Brush.verticalGradient(
                            colors = listOf(
                                MaterialTheme.colorScheme.primary.copy(alpha = 0.08f),
                                MaterialTheme.colorScheme.surface
                            )
                        )
                    )
            )
            
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .verticalScroll(rememberScrollState())
                    .padding(horizontal = 16.dp, vertical = 24.dp)
            ) {
                // Header Section
                HeaderSection()
                
                Spacer(modifier = Modifier.height(32.dp))
                
                // History Section
                HistorySection()
                
                Spacer(modifier = Modifier.height(32.dp))
                
                // How It Works Section
                HowItWorksSection()
                
                Spacer(modifier = Modifier.height(32.dp))
                
                // Benefits Section
                BenefitsSection()
                
                Spacer(modifier = Modifier.height(32.dp))
                
                // Considerations Section
                ConsiderationsSection()
                
                Spacer(modifier = Modifier.height(32.dp))
                
                // CTA Section
                CTASection(onGetStartedClick = onGetStartedClick)
                
                Spacer(modifier = Modifier.height(32.dp))
            }
        }
    }
}

@Composable
private fun HeaderSection() {
    Column(
        modifier = Modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        // Timer Icon
        Box(
            modifier = Modifier
                .size(100.dp)
                .clip(CircleShape)
                .background(MaterialTheme.colorScheme.primary.copy(alpha = 0.15f)),
            contentAlignment = Alignment.Center
        ) {
            Text(
                text = "⏲️",
                style = MaterialTheme.typography.displayMedium
            )
        }
        
        Spacer(modifier = Modifier.height(16.dp))
        
        Text(
            text = "The Power of Pomodoro",
            style = MaterialTheme.typography.headlineLarge,
            fontWeight = FontWeight.Bold,
            textAlign = TextAlign.Center
        )
        
        Spacer(modifier = Modifier.height(8.dp))
        
        Text(
            text = "Focus deeper. Work smarter. Rest better.",
            style = MaterialTheme.typography.titleMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = TextAlign.Center
        )
    }
}

@Composable
private fun HistorySection() {
    Column {
        SectionHeader(icon = "📚", title = "The Origin Story")
        
        Spacer(modifier = Modifier.height(16.dp))
        
        InfoCard {
            Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                Text(
                    text = "The Pomodoro Technique was created in the late 1980s by Francesco Cirillo, an Italian university student seeking a better way to study and manage his time.",
                    style = MaterialTheme.typography.bodyMedium
                )
                
                Text(
                    text = "Named after the tomato-shaped kitchen timer (pomodoro means \"tomato\" in Italian) he used to track his work intervals, this simple yet powerful method has since helped millions worldwide achieve better focus and productivity.",
                    style = MaterialTheme.typography.bodyMedium
                )
                
                Row(
                    modifier = Modifier.padding(top = 4.dp),
                    horizontalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    Text(
                        text = "💡",
                        style = MaterialTheme.typography.bodyMedium
                    )
                    Text(
                        text = "Born from necessity, perfected through practice",
                        style = MaterialTheme.typography.bodyMedium,
                        fontStyle = FontStyle.Italic,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }
        }
    }
}

@Composable
private fun HowItWorksSection() {
    Column {
        SectionHeader(icon = "⚙️", title = "How It Works")
        
        Spacer(modifier = Modifier.height(8.dp))
        
        Text(
            text = "The technique is simple, sustainable, and scientifically backed:",
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
        
        Spacer(modifier = Modifier.height(12.dp))
        
        Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
            StepCard(
                number = 1,
                title = "Choose Your Task",
                description = "Select a single task or project you want to focus on.",
                icon = "✓",
                color = MaterialTheme.colorScheme.primary
            )
            
            StepCard(
                number = 2,
                title = "Set the Timer",
                description = "Start a 25-minute focus session (one \"pomodoro\").",
                icon = "⏲️",
                color = MaterialTheme.colorScheme.primary
            )
            
            StepCard(
                number = 3,
                title = "Work Without Distraction",
                description = "Focus entirely on your task until the timer rings.",
                icon = "🧠",
                color = MaterialTheme.colorScheme.primary
            )
            
            StepCard(
                number = 4,
                title = "Take a Short Break",
                description = "Rest for 5 minutes. Stretch, hydrate, or relax.",
                icon = "☕",
                color = Color(0xFF4CAF50)
            )
            
            StepCard(
                number = 5,
                title = "Repeat & Recharge",
                description = "After 4 pomodoros, take a longer 15-30 minute break.",
                icon = "🔄",
                color = Color(0xFF2196F3)
            )
        }
    }
}

@Composable
private fun BenefitsSection() {
    Column {
        SectionHeader(icon = "⭐", title = "Why It Works")
        
        Spacer(modifier = Modifier.height(8.dp))
        
        Text(
            text = "Research and millions of users have found these key benefits:",
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
        
        Spacer(modifier = Modifier.height(12.dp))
        
        Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
            BenefitCard(
                icon = "👁️",
                title = "Enhanced Focus",
                description = "Short, timed sessions help maintain deep concentration and prevent mental drift.",
                color = MaterialTheme.colorScheme.primary
            )
            
            BenefitCard(
                icon = "🔋",
                title = "Reduced Mental Fatigue",
                description = "Regular breaks prevent burnout and keep your mind fresh throughout the day.",
                color = Color(0xFF4CAF50)
            )
            
            BenefitCard(
                icon = "⏰",
                title = "Better Time Awareness",
                description = "Learn to accurately estimate how long tasks take and plan more effectively.",
                color = Color(0xFF2196F3)
            )
            
            BenefitCard(
                icon = "📈",
                title = "Increased Motivation",
                description = "Small wins and completed sessions create positive momentum and build confidence.",
                color = Color(0xFF9C27B0)
            )
            
            BenefitCard(
                icon = "🚶",
                title = "Sustainable Work Rhythm",
                description = "Balance focused work with restorative breaks for long-term productivity.",
                color = Color(0xFF00BCD4)
            )
            
            BenefitCard(
                icon = "🛑",
                title = "Distraction Management",
                description = "The commitment to 25 minutes helps you defer interruptions and stay on track.",
                color = Color(0xFFFF9800)
            )
        }
    }
}

@Composable
private fun ConsiderationsSection() {
    Column {
        SectionHeader(icon = "⚠️", title = "Things to Keep in Mind")
        
        Spacer(modifier = Modifier.height(16.dp))
        
        InfoCard {
            Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                ConsiderationRow(
                    icon = "🔀",
                    text = "May interrupt deep creative flow states that benefit from longer unbroken periods."
                )
                
                HorizontalDivider()
                
                ConsiderationRow(
                    icon = "📅",
                    text = "The rigid structure might not suit all task types or work environments."
                )
                
                HorizontalDivider()
                
                ConsiderationRow(
                    icon = "👥",
                    text = "Collaborative work may require flexibility with timing to accommodate others."
                )
                
                HorizontalDivider()
                
                ConsiderationRow(
                    icon = "🎚️",
                    text = "Feel free to adjust session lengths to find what works best for you!"
                )
            }
        }
        
        Spacer(modifier = Modifier.height(12.dp))
        
        Text(
            text = "The Pomodoro Technique is a tool, not a rule. Adapt it to your needs and workflow.",
            style = MaterialTheme.typography.bodySmall,
            fontStyle = FontStyle.Italic,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = Modifier.padding(horizontal = 4.dp)
        )
    }
}

@Composable
private fun CTASection(onGetStartedClick: () -> Unit) {
    Column(
        modifier = Modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = "Ready to experience the benefits?",
            style = MaterialTheme.typography.titleLarge,
            fontWeight = FontWeight.SemiBold,
            textAlign = TextAlign.Center
        )
        
        Spacer(modifier = Modifier.height(16.dp))
        
        Button(
            onClick = onGetStartedClick,
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 8.dp),
            shape = RoundedCornerShape(12.dp),
            contentPadding = PaddingValues(vertical = 16.dp)
        ) {
            Icon(
                imageVector = Icons.Default.PlayArrow,
                contentDescription = null,
                modifier = Modifier.size(20.dp)
            )
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = "Start Your First Pomodoro",
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.SemiBold
            )
        }
    }
}

// Supporting Composables

@Composable
private fun SectionHeader(icon: String, title: String) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Text(
            text = icon,
            style = MaterialTheme.typography.headlineSmall
        )
        Text(
            text = title,
            style = MaterialTheme.typography.headlineSmall,
            fontWeight = FontWeight.Bold
        )
    }
}

@Composable
private fun InfoCard(content: @Composable () -> Unit) {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        tonalElevation = 2.dp,
        shadowElevation = 4.dp
    ) {
        Box(modifier = Modifier.padding(16.dp)) {
            content()
        }
    }
}

@Composable
private fun StepCard(
    number: Int,
    title: String,
    description: String,
    icon: String,
    color: Color
) {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        tonalElevation = 1.dp,
        shadowElevation = 2.dp
    ) {
        Row(
            modifier = Modifier.padding(12.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            // Number circle
            Box(
                modifier = Modifier
                    .size(40.dp)
                    .clip(CircleShape)
                    .background(color.copy(alpha = 0.15f)),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = number.toString(),
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = color
                )
            }
            
            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                Row(
                    horizontalArrangement = Arrangement.spacedBy(6.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(text = icon, style = MaterialTheme.typography.bodyMedium)
                    Text(
                        text = title,
                        style = MaterialTheme.typography.titleSmall,
                        fontWeight = FontWeight.SemiBold
                    )
                }
                Text(
                    text = description,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

@Composable
private fun BenefitCard(
    icon: String,
    title: String,
    description: String,
    color: Color
) {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        tonalElevation = 1.dp,
        shadowElevation = 2.dp
    ) {
        Row(
            modifier = Modifier.padding(12.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            // Icon box
            Box(
                modifier = Modifier
                    .size(44.dp)
                    .clip(RoundedCornerShape(10.dp))
                    .background(color.copy(alpha = 0.15f)),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = icon,
                    style = MaterialTheme.typography.titleMedium
                )
            }
            
            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.titleSmall,
                    fontWeight = FontWeight.SemiBold
                )
                Text(
                    text = description,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

@Composable
private fun ConsiderationRow(icon: String, text: String) {
    Row(
        horizontalArrangement = Arrangement.spacedBy(12.dp),
        verticalAlignment = Alignment.Top
    ) {
        Text(
            text = icon,
            style = MaterialTheme.typography.bodyMedium,
            modifier = Modifier.width(24.dp)
        )
        Text(
            text = text,
            style = MaterialTheme.typography.bodyMedium,
            modifier = Modifier.weight(1f)
        )
    }
}

@Preview(showBackground = true)
@Composable
private fun PomodoroBenefitsScreenPreview() {
    PomodoroTheme {
        PomodoroBenefitsScreen(
            onBackClick = {},
            onGetStartedClick = {}
        )
    }
}

@Preview(showBackground = true, uiMode = android.content.res.Configuration.UI_MODE_NIGHT_YES)
@Composable
private fun PomodoroBenefitsScreenDarkPreview() {
    PomodoroTheme {
        PomodoroBenefitsScreen(
            onBackClick = {},
            onGetStartedClick = {}
        )
    }
}
